### DISCUSSION TITLE
[Q/A]How does provider response attachments to consumer in Dubbo2.7.18?

### DISCUSSION THREAD

**Discussion by LiebeKrake** [apache/dubbo#13918] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

provider：
public class MyGenericService implements GenericService {
    public Object $invoke(String methodName, String[] parameterTypes, Object[] args) throws GenericException {
RpcContext serverContext = RpcContext.getServerContext();
serverContext.setAttachment("key","value");
return "xxxx";
}

consumer:
...
registryConfig.setAddress("N/A");
...
referenceConfig.setUrl("dubbo://localhost:8888");
referenceConfig.get().$invoke("echo", new String[]{"java.lang.String"}, new String[]{body});
RpcContext.getServerContext().getAttachment("key");

——————
My code snippet is shown above,
I cannot get the value of  the attachment "key" on the consumer side. How should I return attachments from the provider to the consumer?
(For some special reasons, only dubbo2 can be used for the time being)

在consumer端拿不到的“key”的值，请问应该怎么在provider端向consumer端回传隐式参数？
（因为某些特殊原因，暂时只能使用dubbo2）

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
Please re-submit the discussion in English :)
